cosmos/lib/cosmos/models/target_model.rb

```diff
@@ -389,14 +389,18 @@ def deploy(gem_path, variables)
           FileUtils.mkdir_p(File.dirname(local_path))
           File.open(local_path, 'wb') { |file| file.write(data) }
           found = true
-          rubys3_client.put_object(bucket: 'config', key: key, body: data)
+          rubys3_client.put_object(bucket: 'config', key: key, body: data) unless validate_only
         end
         raise "No target files found at #{target_path}" unless found
 
         target_folder = File.join(temp_dir, @name)
-        build_target_archive(rubys3_client, temp_dir, target_folder)
-        system = update_store(temp_dir)
-        deploy_microservices(gem_path, variables, system)
+        # Build a System for just this target
+        system = System.new([@name], temp_dir)
```

Comment:
This is where most of the validation takes place ... all the targets and their cmd/tlm